pkg/collect/host_run.go

```
@@ -118,6 +136,9 @@ func (c *CollectHostRun) Collect(progressChan chan<- interface{}) (map[string][]
 		if werr, ok := err.(*exec.ExitError); ok {
 			runInfo.ExitCode = strings.TrimPrefix(werr.Error(), "exit status ")
 			runInfo.Error = stderr.String()
+		} else if err == context.DeadlineExceeded {
+			runInfo.ExitCode = "0"
```

Should't we use `werr.ExitCode()`? I'd prefer not "swallowing" the error code.

---

There is no Exit Code as as the error is about context becomes done prior to the command is run. Do you think we should update the artificial Exit Code to something else? E.g. `124`?

---

The value will be `-1` which the [std library uses to indicate](https://cs.opensource.google/go/go/+/refs/tags/go1.21.6:src/os/exec_posix.go;l=133) that the process did not complete. Any other value can be misinterpreted as a valid return code from the process.

---

many thanks @banjoh this is done.